### DISCUSSION TITLE
Package request dict-like object list of packages

### DISCUSSION THREAD

**Discussion by hughetop** [AcademySoftwareFoundation/rez#1344] [New Features]
Hello,

For the [request object](https://github.com/AcademySoftwareFoundation/rez/wiki/Package-Commands#request) we can do things like:
```python
@late()
def requires():
    packages = []
    if 'PySide2' not in request:
        packages.append('PySide2')
    return packages
```
This is great, but if you have multiple things you want to check, you need to do each one individually like:
```python
if 'PyQt4' not in request and 'PyQt5' not in request and 'PySide' not in request and 'PySide2' not in request:
    packages.append('PySide2')
```
Would it be possible to get the list or set of package requests? The [request base class](https://github.com/AcademySoftwareFoundation/rez/blob/master/src/rez/rex_bindings.py#L160) only implements `__contains__` and `__getitem__`. Ideally we could do something like:
```python
if not request.intersection({'PyQt4', 'PyQt5', 'PySide', 'PySide2'}):
    packages.append('PySide2')
```

**Comment by JeanChristopheMorinPerso** [comment]
Something similar has been tried, but it was iterating over `request`, see https://github.com/AcademySoftwareFoundation/rez/pull/608. The `request` object definitely need some enhancements to really make it dict-like. Right now it's partly dict-like.

**Reply by hughetop** [reply]
Ah, interesting. If the request object had a way to get a copy of the packages, like `set(self._data.values())` that could work for iteration and package intersection.
